src/jobs/extraction_jobs.py

```diff
@@ -0,0 +1,237 @@
+﻿"""
+RQ job implementations for PDF extraction using PyMuPDF.
+
+This module contains the actual job functions that are executed by RQ workers.
+Following the rq_pymupdf reference implementation pattern for clean separation
+between Apache 2.0 orchestration code and AGPL 3.0 PyMuPDF processing code.
+"""
+
+import os
+import time
+import tempfile
+from typing import Dict, Any, Optional, Tuple, List
+
+# Import the existing extraction logic (AGPL code)
+from ..extract import extract_markdown_with_hierarchy, ExtractionConfig
+
+
+def extract_pdf_markdown_job(
```

Comment:
you'd need to add the `@job(queue='pdf_queue', timeout=300, result_ttl=3600)` function decorator for rq to call this function and have it be reference-able